database-plugins/src/main/java/io/cdap/plugin/DBRecord.java

```diff
@@ -184,32 +184,32 @@ private void setField(ResultSet resultSet, StructuredRecord.Builder recordBuilde
       bytesRead += decimal.unscaledValue().bitLength() / Byte.SIZE + Integer.BYTES;
       recordBuilder.setDecimal(field.getName(), decimal);
     } else {
-      Schema schema = field.getSchema();
-      if (schema.isNullable()) {
-        schema = schema.getNonNullable();
-      }
-      switch (schema.getType()) {
-        case INT:
-          bytesRead += Integer.BYTES;
-          break;
-        case LONG:
-          bytesRead += Long.BYTES;
-          break;
-        case DOUBLE:
-          bytesRead += Double.BYTES;
-          break;
-        case FLOAT:
-          bytesRead += Float.BYTES;
-          break;
-        case STRING:
-          bytesRead += ((String) o).length();
-          break;
-        case BYTES:
-          bytesRead += ((byte[]) o).length;
-          break;
-        case BOOLEAN:
-          bytesRead += Integer.BYTES;
-          break;
+      if (o != null) {
```

Comment:
nit: combine this with the else on the line above so it would be `else if (o != null){ ...`? Might make it easier to read

Reply:
Can't combine it because we still need to execute `recordBuilder.set(field.getName(), o);` in [line 214](https://github.com/cdapio/hydrator-plugins/pull/1322/files#diff-ba8230c85bf223ca454f424bba027c5712f77202d54945939d6f01de49a0a7faR214)